sdk/core/core-http/src/serializer.ts

```diff
@@ -918,6 +919,7 @@ function deserializeCompositeType(
             propertyObjectName,
             options
           );
+          handledPropertyNames.push(xmlName!);
```

Comment:
Do we need to do similar special handling for this case?

```
     if (propertyMapper.xmlIsAttribute && responseBody[XML_ATTRKEY]) {
        instance[key] = serializer.deserialize(
          propertyMapper,
          responseBody[XML_ATTRKEY][xmlName!],
          propertyObjectName,
          options
        );
      }
```

that is to say if the xmlName of the attribute is different than the serialized name for some reason, might we have the same problem with not removing it? Or do attributes not suffer this same problem?

Reply:
Good point! Yes we should do the same there too.

Reply:
Actually when `xmlIsAttribute` is true, we are reading it from the "$" property, and we don't copy the "$" or "_", so should be fine.